src/module.c

```diff
@@ -8659,6 +8737,43 @@ int RM_SetModuleUserACL(RedisModuleUser *user, const char* acl) {
     return ACLSetUser(user->user, acl, -1);
 }
 
+sds addACLsToUser(user *u, sds username, sds *argv, int argc);
+
+RedisModuleString * RM_SetModuleUserACLString(RedisModuleCtx * ctx, RedisModuleUser *user, const char* acl) {
+    sds sacl = sdsnew(acl);
+    int argc;
+    sds *argv = sdssplitargs(sacl, &argc);
+
+    sds error = addACLsToUser(user->user, NULL, argv, argc);
+
+    sdsfreesplitres(argv, argc);
+    sdsfree(sacl);
+
+    if (error) {
```

Comment:
I'm not sure it is required to propagate the error...

Reply:
if we fail to parse the string, isn't it import to know why?  the other option would be for it not to be a return value but for the fucn to take a RedisModule ** and return it that way.

Reply:
so, what I think the answer is here, is to pass in a RedisModuleString ** to the function.  If it's not null, we generate a string, otherwise we don't. and then return value is just REDISMOULE_OK/ERR and if caller wants a string with the error, they'll get it.